src/resource_manager/gateway.py

```diff
@@ -62,29 +79,40 @@ def _label_selector(self) -> str:
         return ",".join(f"{k}={v}" for k, v in self._labels.items())
 
     @map_k8s_auth_exception
-    def _gen_resource(self, definition: GatewayResourceDefinition, config: CharmConfig) -> dict:
+    def _gen_resource(self, state: State) -> dict:
```

Comment:
As mentioned in another comment, we should probably rename this to `ResourceDefinition` and `resource_definition`

Reply:
I renamed this to ResourceDefinition as it indeed makes more sense